source/common/upstream/load_balancer_impl.cc

```diff
@@ -724,20 +748,42 @@ void EdfLoadBalancerBase::initialize() {
   }
 }
 
+void EdfLoadBalancerBase::recalculateHostsInSlowStart(const HostVector& hosts) {
+  auto current_time = time_source_.monotonicTime();
```

Comment:
now as hosts_in_slow_start (with new comparator) support removal by single host, could process hosts_removed and remove those from slow start collection.